sklearnex/cluster/k_means.py

```diff
@@ -12,6 +12,412 @@
+                    max_iter=max_iter,
+                    tol=tol,
+                    precompute_distances=precompute_distances,
+                    n_init=n_init,
+                    verbose=verbose,
+                    random_state=random_state,
+                    copy_x=copy_x,
+                    n_jobs=n_jobs,
+                    algorithm=algorithm,
+                )
+
+        def _initialize_onedal_estimator(self):
+            onedal_params = {
+                "n_clusters": self.n_clusters,
+                "init": self.init,
+                "max_iter": self.max_iter,
+                "tol": self.tol,
+                "n_init": self.n_init,
+                "verbose": self.verbose,
+                "random_state": self.random_state,
+            }
+
+            self._onedal_estimator = onedal_KMeans(**onedal_params)
+
+        def fit(self, X, y=None, sample_weight=None):
+            """Compute k-means clustering.
+
+            Parameters
+            ----------
+            X : array-like or sparse matrix, shape=(n_samples, n_features)
+                Training instances to cluster. It must be noted that the data
+                will be converted to C ordering, which will cause a memory
+                copy if the given data is not C-contiguous.
+
+            y : Ignored
+                not used, present here for API consistency by convention.
+
+            sample_weight : array-like, shape (n_samples,), optional
+                The weights for each observation in X. If None, all observations
+                are assigned equal weight (default: None)
+
+            """
+
+            if sklearn_check_version("1.0"):
+                self._check_feature_names(X, reset=True)
+            if sklearn_check_version("1.2"):
+                self._validate_params()
+
+            dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_KMeans.fit,
+                },
+                X,
+                y,
+                sample_weight,
+            )
+
+            return self
+
+        @run_with_n_jobs
+        def _onedal_fit(self, X, _, sample_weight, queue=None):
+            assert sample_weight is None
+
+            X = self._validate_data(
+                X,
+                accept_sparse=False,
+                dtype=[np.float64, np.float32],
+                order="C",
+                copy=self.copy_x,
+            )
+
+            if sklearn_check_version("1.2"):
+                self._check_params_vs_input(X)
+            else:
+                self._check_params(X)
+
+            self._n_features_out = self.n_clusters
+            self._n_threads = _openmp_effective_n_threads()
+
+            self._initialize_onedal_estimator()
+            self._onedal_estimator.fit(X, queue=queue)
+
+            self._save_attributes()
+
+        @wrap_output_data
+        def predict(self, X):

```

Comment:
This should also fail for a test looking for signature matching of methods. Again, something is not occurring properly in CI.